### PR TITLE
[diligent-core] add version 250014

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **diligent/250014**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
I have the following error locally:

CMake Error at CMakeLists.txt:7 (find_package):
  Could not find a package configuration file provided by "xxHash" with any
  of the following names:

    xxHashConfig.cmake
    xxhash-config.cmake


why does it happen? I have copied this line from xhash test package:
find_package(xxHash REQUIRED CONFIG)
why it does not work in my case?

**Comment by AndreyMlashkin** [comment]
@ericLemanissier @prince-chrismc @SSE4, could someone help, please?

**Comment by SSE4** [comment]
> @ericLemanissier @prince-chrismc @SSE4, could someone help, please?

I am taking a look

**Comment by SSE4** [comment]
`find_package(xxHash REQUIRED CONFIG)` - this line searches for the CMake config files only (`xxHashConfig.cmake` or `xxhash-config.cmake`).
while recipe used `cmake_find_package` generator, which generates find modules (`FindxxHash.cmake`).
I've changed recipe to use `cmake_find_package_multi` instead (as config modules are recommended over legacy find modules).
it allowed build to proceed further, but it failed at test package for me:
```
C:\.conan\a6dcee\1\source_subfolder\Graphics\ShaderTools\src\GLSLangUtils.cpp(445,12): error C2039: 'setDxPositionW': is not a member of 'glslang::T
Shader' [C:\.conan\a6dcee\1\build_subfolder\source_subfolder\Graphics\ShaderTools\Diligent-ShaderTools.vcxproj]
```

perhaps, interface has changed in this new version.

**Comment by AndreyMlashkin** [comment]
I have no clue, why it is compiled with gcc13 on my machine, but it fails with gcc10...

**Comment by AndreyMlashkin** [comment]
Could please someone help with this one? It seems wired for me, that such an error is depending on compiler or cmake version

**Comment by SSE4** [comment]
> Could please someone help with this one? It seems wired for me, that such an error is depending on compiler or cmake version

as I can see, the project that doesn't see vulkan headers is `Graphics/Archiver`. perhaps, we should inject conan vulka headers the same as we did for `Graphics/GraphicsEngineVulkan`? (https://github.com/conan-io/conan-center-index/pull/9631/files#diff-18926468eddeb38ccf631aa2c716efc2e0e75baa6393efe84bf9babbe6eeafb0R10).

**Comment by ericLemanissier** [comment]
https://github.com/DiligentGraphics/DiligentCore/archive/refs/tags/API250014.tar.gz does not have vulkan.h, contrary to https://github.com/DiligentGraphics/DiligentCore/archive/refs/tags/v2.5.1.tar.gz
Are we sure we want to package this API*** tag?

**Comment by AndreyMlashkin** [comment]
Thanks, seems to work :)

**Comment by AndreyMlashkin** [comment]
This API tag is newer then 2.5.1
There is no any logic, but it is so...

**Comment by ericLemanissier** [comment]
more or less repeating what I already said, but API250014 is not an actual release of diligent-core, but a label of the api. Please remove this tag from your PR.

**Comment by AndreyMlashkin** [comment]
> more or less repeating what I already said, but API250014 is not an actual release of diligent-core, but a label of the api. Please remove this tag from your PR.

Why? It's the last created tag. Tags are always preferred over the cci version, aren't they?

https://github.com/DiligentGraphics/DiligentCore/issues/355

**Comment by AndreyMlashkin** [comment]
I wasn't able to reproduce the error on my windows machine...
any hint from CI experts?

@jgsogo

**Comment by prince-chrismc** [comment]
> I wasn't able to reproduce the error on my windows machine...
> any hint from CI experts?

No expert either but I was able to repo the problem locally...

The failing code https://github.com/DiligentGraphics/DiligentCore/blob/77dfc259d70bd3ab44fed7696eb582bbb26368fb/BuildUtils.cmake#L314-L319 looked odd since you cant have generator expressions

or so I though

> New in version 3.20: One may use generator expressions to specify per-configuration outputs. For example, the code:

https://cmake.org/cmake/help/latest/command/add_custom_command.html#examples-generating-files

So I upgraded CMake 

```
C:\Users\Chris McArthur\Documents\conan-center-index\recipes\diligent-core>cmake --version
cmake version 3.18.4

CMake suite maintained and supported by Kitware (kitware.com/cmake).

C:\Users\Chris McArthur\Documents\conan-center-index\recipes\diligent-core>cmake --version
cmake version 3.22.3
```


This however still failed 

```
C:\.conan\d17aa6\1\source_subfolder\Graphics\ShaderTools\src\DXBCUtils.cpp(32,10): fatal error C1083: Cannot open include file: 'atlcomcli.h': No such file or directory [C:\.
conan\d17aa6\1\build_subfolder\source_subfolder\Graphics\ShaderTools\Diligent-ShaderTools.vcxproj]
C:\.conan\d17aa6\1\source_subfolder\Graphics\ShaderTools\src\GLSLangUtils.cpp(36,1): warning C4005: 'ENABLE_HLSL': macro redefinition [C:\.conan\d17aa6\1\build_subfolder\sour
ce_subfolder\Graphics\ShaderTools\Diligent-ShaderTools.vcxproj]
C:\.conan\d17aa6\1\source_subfolder\Graphics\ShaderTools\src\GLSLangUtils.cpp : message : see previous definition of 'ENABLE_HLSL' [C:\.conan\d17aa6\1\build_subfolder\source_
subfolder\Graphics\ShaderTools\Diligent-ShaderTools.vcxproj]
C:\.conan\d17aa6\1\source_subfolder\Graphics\ShaderTools\include\DXCompilerBaseWin32.hpp(32,10): fatal error C1083: Cannot open include file: 'atlbase.h': No such file or dir
ectory (compiling source file C:\.conan\d17aa6\1\source_subfolder\Graphics\ShaderTools\src\DXCompiler.cpp) [C:\.conan\d17aa6\1\build_subfolder\source_subfolder\Graphics\Shade
rTools\Diligent-ShaderTools.vcxproj]
```

But that looks like missing compiler features so hopefully CCI wont have that problem 🤞

**Comment by prince-chrismc** [comment]
Trying adding a build requirement for CMake to something more recent on Windows and I think we should have better results

**Comment by AndreyMlashkin** [comment]
@prince-chrismc, thanks a lot! That means, cmake version should be updated in CI! @jgsogo 

and about the error
Cannot open include file: 'atlcomcli.h'
MSVC should have atl and mfc components installed in order to compile it. Standard installation of MSVC-buildtools does not have it.

**Comment by AndreyMlashkin** [comment]
Finally! Please approve

@uilianries @SSE4 @ericLemanissier @prince-chrismc @jgsogo

**Review by SSE4 — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]

File: recipes/diligent-core/all/conanfile.py [hunk 1]
```diff
@@ -58,6 +58,8 @@ def validate(self):
                     self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
         if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime:
             raise ConanInvalidConfiguration("Visual Studio build with MT runtime is not supported")
+        if self.version == "250014" and self.settings.compiler == 'Visual Studio' and self.settings.build_type == 'Debug':
+            raise ConanInvalidConfiguration("Can't build Visual Studio in Debug mode")
```
Comment (prince-chrismc) [inline recipes/diligent-core/all/conanfile.py:62]: Can we try removing this? I had the impression it was only a test but it should pass 🤞 given my result locally

```suggestion
```

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — DISMISSED** [review]
The tag is not an official release, but it works better than cci custom tags, because comes the upstream.

**Review by jgsogo — COMMENTED** [review]
When compared using semver (loose), 250014 will be greater than any other release, right? Is this what we want? Does Conan resolve version ranges using loose by default?

**Comment by SSE4** [comment]
> When compared using semver (loose), 250014 will be greater than any other release, right? Is this what we want? Does Conan resolve version ranges using loose by default?

yes, I think latest will always resolve to 250014 instead of 2.5.x

**Review by jgsogo — CHANGES_REQUESTED** [review]
I'm not sure this is something we want (given that there are actual versions). Please, share your thoughts

**Comment by SSE4** [comment]
> Alternative would be to use `API250014`

yes, I support it, use exactly the same version as a tag name, with `api` prefix. it will ensure version ranges won't resolve to that `api` tags.

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — DISMISSED** [review]

**Review by SSE4 — DISMISSED** [review]

**Comment by AndreyMlashkin** [comment]
@ericLemanissier @prince-chrismc @SSE4 @uilianries @jgsogo
Please approve

**Review by SSE4 — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
@ericLemanissier @prince-chrismc@uilianries @jgsogo
please approve

**Comment by AndreyMlashkin** [comment]
Could you please speed up the review? It's already pending for a week. There are already other pull request, that will cause a rebase ...

**Review by uilianries — APPROVED** [review]

**Comment by SSE4** [comment]
> Could you please speed up the review? It's already pending for a week. There are already other pull request, that will cause a rebase ...

I wish I could. maybe it's time to add more community reviewers @jgsogo @danimtb

**Review by jgsogo — APPROVED** [review]
Yes, we need to add more reviewers. ATM I haven´t enough time to review PRs here 😞

**Pull request merged**